### ISSUE TITLE
Build failure: ipu6-drivers

### ISSUE THREAD

**Issue by imchairmanm** [NixOS/nixpkgs#303067]
### Steps To Reproduce

1. build linuxKernel.packages.linux_6_8.ipu6-drivers on unstable

### Build log

https://gist.github.com/imchairmanm/09df2e3fe9b38c23b79b9029213a86c5

### Additional context

Not sure when the break occurred, but this stopped working at some point.

The failure occurs when I have this in my configuration.nix file:

```
hardware.ipu6.enable = true;
hardware.ipu6.platform = "ipu6ep";
```

I'm on a X1 Carbon gen 10.  Not sure if it's relevant, but I filed a [similar build failure](https://github.com/NixOS/nixpkgs/issues/267416) previously for a previous version if that's helpful context.

### Notify maintainers

@mweinelt

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9, NixOS, 24.05 (Uakari), 24.05.20240408.4cba8b5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/450afzqlzzgw6wnyc3dwysf3i5yxyqkr-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by vale981** [comment]
same with kernel 6.9

**Comment by matejc** [comment]
Related? https://github.com/intel/ipu6-drivers/issues/232

**Comment by adminy** [comment]
Just did a fresh build, it looks like some v4l functions got renamed:
https://logpaste.com/LYmKBUWK

**Comment by wegank** [comment]
Closed by #331747 for Linux 6.10+; Linux 6.7-6.9 have been removed from Nixpkgs.
